spec/lucky/param_parser_spec.cr

```diff
@@ -0,0 +1,127 @@
+require "../spec_helper"
+
+enum TimeComponent
+  Year
+  Month
+  Day
+  Hour
+  Minute
+  Second
+end
+
+struct FormattedTime
```

Comment:
All of this was directly taken from Avram https://github.com/luckyframework/avram/blob/9a5a921bb05cc3031d9436d2fdc6927606314876/spec/type_extensions/time_spec.cr#L3-L27

@stephendolan saved me from having to figure out all these tests myself 😅